pyretailscience/plots/cohort.py

```diff
@@ -0,0 +1,96 @@
+- **Customizable Labels**: Supports custom labels for x-axis, y-axis, title, and colorbar.
+- **Source Text**: Provides an option to add source attribution to the plot.
+- **Grid and Tick Customization**: Applies standard styling for better readability.
+
+### Use Cases
+
+- **Cohort Analysis**: Visualize how different groups behave over time.
+- **Category-Based Heatmaps**: Compare values across different categories.
+
+### Limitations and Warnings
+
+- **Data Aggregation Required**: The module does not perform data aggregation; data should be pre-aggregated before being passed to the function.
+- **Fixed Color Mapping**: The module uses a predefined colormap without dynamic adjustments.
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+from matplotlib import ticker
+from matplotlib.axes import Axes, SubplotBase
+
+import pyretailscience.style.graph_utils as gu
+from pyretailscience.style.tailwind import get_listed_cmap
+
+
+def plot(
+    df: pd.DataFrame,
+    cbarlabel: str,
+    x_label: str | None = None,
+    y_label: str | None = None,
+    title: str | None = None,
+    ax: Axes | None = None,
+    source_text: str | None = None,
+    percentage: bool = False,
+    **kwargs: dict,
+) -> SubplotBase:
+    """Plots a cohort plot for the given DataFrame.
+
+    Args:
+        df (pd.DataFrame): Dataframe containing cohort analysis data.
+        cbarlabel (str): Label for the colorbar.
+        x_label (str, optional): Label for x-axis.
+        y_label (str, optional): Label for y-axis.
+        title (str, optional): Title of the plot.
+        ax (Axes, optional): Matplotlib axes object to plot on.
+        source_text (str, optional): Additional source text annotation.
+        percentage (bool, optional): If True, displays cohort values as percentages. Defaults to False.
+        **kwargs: Additional keyword arguments for cohort styling.
+
+    Returns:
+        SubplotBase: The matplotlib axes object.
+    """
+    ax = ax or plt.gca()
+    cmap = get_listed_cmap("green")
+    im = ax.imshow(df, cmap=cmap, **kwargs)
+    cbar = ax.figure.colorbar(im, ax=ax)
+    cbar.ax.set_ylabel(cbarlabel, rotation=-90, va="bottom", fontsize="x-large")
+
+    ax.set_xticks(np.arange(df.shape[1]))
+    ax.set_yticks(np.arange(df.shape[0]))
+    ax.set_xticklabels(df.columns, rotation=-30, ha="right", rotation_mode="anchor")
+    ax.set_yticklabels(df.index.astype(str))
+
+    ax.tick_params(top=True, bottom=False, labeltop=True, labelbottom=False)
+    ax.set_xticks(np.arange(df.shape[1] + 1) - 0.5, minor=True)
+    ax.set_yticks(np.arange(df.shape[0] + 1) - 0.5, minor=True)
+    ax.grid(which="minor", color="w", linestyle="-", linewidth=3)
+    ax.tick_params(which="minor", bottom=False, left=False)
+    threshold = im.norm(1.0) / 2.0 if percentage else im.norm(df.to_numpy().max()) / 2.0
+    valfmt = ticker.StrMethodFormatter("{x:.0%}" if percentage else "{x:,.0f}")
```

Comment:
Can you make it so the user can pass this in as a parameter. Sometimes you need more/fewer decimals. eg
```suggestion
    if number_format is None:
        valfmt = ticker.StrMethodFormatter("{x:.0%}" if percentage else "{x:,.0f}")
    else:
        valfmt = ticker.StrMethodFormatter(number_format)
```